src/Ast/PhpClass.php

```diff
@@ -28,13 +28,18 @@ final class PhpClass extends AST
 {
     use HasPhpDoc;
 
-    public function __construct(Type $type, ?ResolvedType $extends, bool $final)
-    {
+    public function __construct(
+        Type $type,
+        ?ResolvedType $extends,
+        bool $final,
```

Comment:
Should this be by default `false`?

Reply:
Thought so too. But,
1. It defaults to `false` in `AST.php`
2. I didn't wanna change it before confirming.

Perhaps @bshaffer can suggest

Reply:
It's required, so I think that's better than having a default

Reply:
The reason that the constructor has a required param is because the main entry point for `PhpClass` creation is the `AST::class` method, which has optional params. The *only* instance of `new PhpClass` is in `AST::class`. As such, I'd prefer the constructor params to be required for the time being, so please change `bool $abstract` to required.

> We could also consider making this an enum:

I like this idea, though I think the parameter for it would need to be an array of `ClassModifier`s(?) so that it could be [`Static`, `Final`] for example. We'd need to also validate which options can be used together of course.

Reply:
That makes sense! I think the only class modifiers are abstract and final, and can't be used together. But yes, an enum array would also work! In a separate PR would be better though.

Reply:
> in a separate PR would be better though.

Yep, for sure.

Reply:
👍 Will tackle the enums part in another PR.